### PR TITLE
dev-docs: add PR review policy

### PR DESCRIPTION
**Pull request by lucasmoura**
## Proposed Commit Message
dev-docs: add PR review policy

Add document describing how the team handle PR reviews

## Test Steps
Double check to see if the policy is correct

## Desired commit type

 - [ ] Squash and merge: Using the "Proposed Commit Message"
 - [ ] Create a merge commit and use "Proposed Commit Message"
 - [x] Rebase and merge, no merge commit, rely only on existing commit messages

## Checklist

 - [ ] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any integration tests accordingly
 - [x] I have updated or added any documentation accordingly

## Does this PR require extra reviews?

 - [ ] Yes
 - [x] No

### CODE REVIEW

File: dev-docs/references/pr_review_policy.md
Comment (renanrodrigo) [inline dev-docs/references/pr_review_policy.md:6]: I'd specify that the approver himself can merge it

**Review by renanrodrigo — APPROVED** [review]

**Review by orndorffgrant — APPROVED** [review]

File: dev-docs/references/pr_review_policy.md
Comment (lucasmoura) [reply dev-docs/references/pr_review_policy.md:6]: All of those PRs can be merged by anyone in the team after we wait for the right amount of time and have the necessary number of reviews. But I don't know if we should make that explicit on the document.

What do you think @renanrodrigo @orndorffgrant @CalvoM @s-makin ?

Comment (orndorffgrant) [reply dev-docs/references/pr_review_policy.md]: While we're writing things like this down, I think we might as well be explicit - maybe something like:
```diff
 * Simple PRs: PRs that fix typos or perform one line changes require **one** approval from a team
   member and can be merged immediately by the reviewer. 
 * Low complexity PRs: For short PRs we require **one** approval to merge. However, after approving the
   PR we will wait for **one** day before merging it. This will allow other team members to review
-  the PR if they want.
+  the PR if they want. When one day has passed after the approval and there are no new requested changes,
+  any team member can merge.
 * Complex PRs: Minimum of **two** approvals from team members. Additionally, some PRs may also
   require review from members that are not on the team, like SRU reviewers. However, this is context
-  dependent, as some PRs do not need this type of review.
+  dependent, as some PRs do not need this type of review. After there are two team-member approvals
+  additional review feedback has been taken into account, and there are now remaining requested
+  changes, any team member can merge.
```

Comment (s-makin) [reply dev-docs/references/pr_review_policy.md]: I think being explicit is always a good idea. The bit about low-complexity PRs gave me a bit of a pause because it looks like we're conflating size with complexity, but the two may or may not be linked (you can have a short/complex or a large/straightforward one). 

It's also a bit unclear from "some PRs do not need this type of review" what PRs that applies to - e.g. does it refer to just the low-complexity ones, or are there special cases of complex ones that don't require it? How do you know which ones need SRU review? Does the team assign reviewers or is it something the proposer has to consider and know who to request review from?

Comment (lucasmoura) [reply dev-docs/references/pr_review_policy.md]: Yes, this is not related to the size at all, but what the PR is actually changing.

And regarding to complex PRs, those are ones that either change/add a Pro service, such as a daemon or a new systemd unit, or ones that are adding features that are tied to system that our team is not an expert at, adding new APT hooks for example. When that happens, we directly ask for review from developers outside the team.

I think I can add some of those examples on that policy

Comment (lucasmoura) [reply dev-docs/references/pr_review_policy.md]: Actually, I have created a new entry for those PRs were we require review from members outside the team. Let me know what you think @orndorffgrant @s-makin

File: dev-docs/references/pr_review_policy.md
Comment (s-makin) [inline dev-docs/references/pr_review_policy.md:12]:
```suggestion
  and there are no remaining requested changes, any team member can merge.
```

File: dev-docs/references/pr_review_policy.md
Comment (s-makin) [inline dev-docs/references/pr_review_policy.md:14]:
```suggestion
* SRU blocker PRs: Some PRs may also require review from members that are not on the team, like SRU reviewers. This is the case, for example, when adding/creating new systemd units). For those PRs, we will directly ask for feedback from members outside
```

File: dev-docs/references/pr_review_policy.md
Comment (s-makin) [inline dev-docs/references/pr_review_policy.md:18]:
```suggestion
  we not only require at least **two** approvals from team members, but also approvals from every external reviewer we assigned. After the required approvals have been given, the additional review feedback has been taken into account, and there are no requested changes remaining, any team member can merge.
```

**Review by s-makin — APPROVED** [review]

**Comment by s-makin** [comment]
> @s-makin updated

LGTM :)

**Pull request merged**
